lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -760,18 +756,62 @@ public static function get_style_variations( $scope = 'theme' ) {
 			$variation_files = array_merge( $variation_files, $variation_files_parent );
 		}
 		ksort( $variation_files );
+
+		return $variation_files;
+	}
+
+	/**
+	 * Returns the style variations defined by the theme (parent and child).
+	 *
+	 * @since 6.2.0 Returns parent theme variations if theme is a child.
+	 * @since 6.6.0 Added cache and configurable scope parameter to allow
+	 *              filtering theme.json partial files by the scope to
+	 *              which they can be applied e.g. theme vs block etc.
+	 *
+	 * @param string $scope The scope or type of style variation to retrieve e.g. theme, block etc.
+	 * @return array
+	 */
+	public static function get_style_variations( $scope = 'theme' ) {
+		$registry  = WP_Block_Type_Registry::get_instance();
+		$blocks    = $registry->get_all_registered();
+		$theme_dir = get_stylesheet_directory();
+		$locale    = get_locale();
+
+		if (
+			isset( static::$style_variations_cache[ $theme_dir ][ $locale ][ $scope ] ) &&
```

Comment:
Actually, I withdraw my [comment](https://github.com/WordPress/gutenberg/pull/62610#issuecomment-2172613741). 

Many of the Rust variation's block styles seems to be missing, e.g., `"core/calendar"` when the cache is returned here.